### PR TITLE
Laboratory work #2, Bogdanova Valeria - 23FPL1

### PR DESCRIPTION
**Pull request by lerabogdanova**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -19,6 +20,19 @@ def tokenize(text: str) -> list[str] | None:
...
+        return None
+
+    text_prep = ''
+    for symb in text.lower():
+        if symb.isalpha():
+            text_prep += symb
+        else:
+            text_prep += ' '
+            continue
+    tokenized_list = text_prep.split()
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:33]: do not create the variable

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -48,6 +75,22 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
...
+
+    unique_words = []
+    for lst_tokens in documents:
+        for token in lst_tokens:
+            if not token or not isinstance(token, str):
+                return None
+            if token not in unique_words:
+                unique_words += [token]
+            else:
+                continue
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:91]: why do we need this?

File: lab_2_retrieval_w_bm25/main.py [hunk 3]
```diff
@@ -78,6 +134,28 @@ def calculate_idf(vocab: list[str], documents: list[list[str]]) -> dict[str, flo
...
+            return None
+        count = 0
+        for doc in documents:
+            for i in doc:
+                if not i or not isinstance(i, str):
+                    return None
+            if un_word in doc:
+                count += 1
+            else:
+                continue
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:155]: whyyyy?

File: lab_2_retrieval_w_bm25/main.py [hunk 4]
```diff
@@ -93,15 +171,30 @@ def calculate_tf_idf(tf: dict[str, float], idf: dict[str, float]) -> dict[str, f
...
 
 def calculate_bm25(
-    vocab: list[str],
-    document: list[str],
-    idf_document: dict[str, float],
-    k1: float = 1.5,
-    b: float = 0.75,
-    avg_doc_len: float | None = None,
-    doc_len: int | None = None,
+        vocab: list[str],
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:191]: revert

File: lab_2_retrieval_w_bm25/main.py [hunk 5]
```diff
@@ -123,7 +216,7 @@ def calculate_bm25(
 
 
 def rank_documents(
-    indexes: list[dict[str, float]], query: str, stopwords: list[str]
+        indexes: list[dict[str, float]], query: str, stopwords: list[str]
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:219]: revert

File: lab_2_retrieval_w_bm25/main.py [hunk 6]
```diff
@@ -141,14 +234,14 @@ def rank_documents(
...
 def calculate_bm25_with_cutoff(
-    vocab: list[str],
-    document: list[str],
-    idf_document: dict[str, float],
-    alpha: float,
-    k1: float = 1.5,
-    b: float = 0.75,
-    avg_doc_len: float | None = None,
-    doc_len: int | None = None,
+        vocab: list[str],
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:237]: revert

File: lab_2_retrieval_w_bm25/settings.json [hunk 7]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 6
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/settings.json:2]: what about 8?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/main.py [hunk 8]
```diff
@@ -19,6 +20,19 @@ def tokenize(text: str) -> list[str] | None:
...
 
+    if not text or not isinstance(text, str):
+        return None
+
+    text_prep = ''
+    for symb in text.lower():
+        if symb.isalpha():
+            text_prep += symb
+        else:
+            text_prep += ' '
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:31]: string concatination is a bad idea

File: lab_2_retrieval_w_bm25/main.py [hunk 9]
```diff
@@ -19,6 +20,19 @@ def tokenize(text: str) -> list[str] | None:
...
+            continue
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:32]: do you really need this?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — COMMENTED** [review]
Address the previous comments, please

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (lerabogdanova) [reply lab_2_retrieval_w_bm25/main.py:91]: to create a list of unique words, without repetitions

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (lerabogdanova) [reply lab_2_retrieval_w_bm25/main.py:155]: to calculate how many documents contain this word, i don't know how to do it any other way :'(

File: lab_2_retrieval_w_bm25/settings.json (hunk 7 above)
Comment (lerabogdanova) [reply lab_2_retrieval_w_bm25/settings.json:2]: i'll try

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (artyomtugaryov) [reply lab_2_retrieval_w_bm25/main.py:91]: what about set?

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (artyomtugaryov) [reply lab_2_retrieval_w_bm25/main.py:155]: it is ugly, find more elegant way

File: lab_2_retrieval_w_bm25/main.py [hunk 10]
```diff
@@ -122,6 +215,7 @@ def calculate_bm25(
     """
 
 
+
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:218]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: 2024-2-level-labs
Comment (artyomtugaryov) [inline 2024-2-level-labs:1]: I don't have idea how you did it, but please, revert this change.

File: lab_2_retrieval_w_bm25/main.py [hunk 11]
```diff
@@ -19,6 +22,17 @@ def tokenize(text: str) -> list[str] | None:
     In case of corrupt input arguments, None is returned.
     """
 
+    if not text or not isinstance(text, str):
+        return None
+
+    text_prep = []
+    for symb in text.lower():
+        if symb.isalpha():
+            text_prep.append(symb)
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:31]: continue and remove `else`

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:91]: strange code, remove

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
The same comments
